apps/ledger-live-desktop/src/renderer/modals/CreateLocalManifest/index.tsx

```diff
@@ -0,0 +1,382 @@
+        ? (prevState.dapp = { ...DEFAULT_FORM.dapp } as LiveAppManifestDapp)
+        : delete prevState.dapp;
+      return prevState;
+    });
+  };
+
+  const formIsValid = useMemo(() => LiveAppManifestSchema.safeParse(form).success, [form]);
+
+  const handleChange = useCallback((path: string, value: unknown) => {
+    setForm((prevState: LiveAppManifest) => {
+      return updateObjectAtPath({ ...prevState }, path, value);
+    });
+  }, []);
+
+  const submitHandler = (e: KeyboardEvent) => {
+    e.preventDefault();
+    if (e.detail !== 0) {
+      formIsValid && addLocalManifest(form);
+      onClose();
+    }
+  };
+
+  return (
+    <form style={{ width: "90%", margin: "auto" }}>
+      <ModalBody
+        onClose={onClose}
+        title={<Trans i18nKey={`settings.developer.createLocalAppModal.title.create`} />}
+        render={() => (
+          <>
+            <ScrollArea>
+              <Flex height={"65vh"} rowGap={15} flexDirection={"column"}>
+                <Flex margin={"auto"} width={"max-content"} columnGap={2}>
+                  <Text marginBottom={1} marginLeft={1} ff="Inter|Medium" fontSize={4}>
+                    {`Wallet-API`}
+                  </Text>
+                  <Switch isChecked={isDapp} onChange={handleSwitchEthDapp} />
+                  <Text marginBottom={1} marginLeft={1} ff="Inter|Medium" fontSize={4}>
+                    {`ETH Dapp`}
+                  </Text>
+                </Flex>
+
+                {Object.keys(form)
+                  .map(key => key as keyof LiveAppManifestSchemaType)
+                  .map((key, index) => {
+                    if (key === "content") {
+                      return Object.keys(form[key])
+                        .map(contentKey => contentKey as keyof LiveAppManifest["content"])
+                        .map(contentKey => {
+                          const value = form.content[contentKey]?.en;
+
+                          if (value === undefined) return <></>;
+
+                          const parseCheck = LiveAppManifestSchema.shape.content.shape[
+                            contentKey
+                          ].safeParse({ en: value }).success;
+                          const optional =
+                            LiveAppManifestSchema.shape.content.shape[contentKey].isOptional();
+                          const path = `${key}.${contentKey as string}.en`;
+
+                          return (
+                            <FormLiveAppInput
+                              key={contentKey as string}
+                              type={typeof value}
+                              fieldName={contentKey as string}
+                              value={value}
+                              optional={optional}
+                              parseCheck={parseCheck}
+                              path={path}
+                              handleChange={handleChange}
+                            />
+                          );
+                        });
```

Comment:
That could be a bit cleaner to split this in a component for content, it can live in the same file IMO